### ISSUE TITLE
IDLE: Input method error in comment with Korean language

### ISSUE THREAD

**Issue by 13961e4d-17cc-4ee4-91db-63e998679732** [python/cpython#75073]
BPO | [30890](https://bugs.python.org/issue30890)
--- | :---
Nosy | @terryjreedy, @mlouielu
Files | <li>[comments_error.png](https://bugs.python.org/file46998/comments_error.png "Uploaded as image/png at 2017-07-10.06:33:39 by DelTos")</li><li>[main.py](https://bugs.python.org/file47003/main.py "Uploaded as text/plain at 2017-07-11.02:35:23 by DelTos")</li><li>[comparison.png](https://bugs.python.org/file47004/comparison.png "Uploaded as image/png at 2017-07-11.03:21:54 by DelTos")</li>

<sup>*Note: these values reflect the state of the issue at the time it was migrated and might not reflect the current state.*</sup>

<details><summary>Show more details</summary><p>

GitHub fields:
```python
assignee = 'https://github.com/terryjreedy'
closed_at = <Date 2017-07-11.23:25:15.855>
created_at = <Date 2017-07-10.06:33:39.665>
labels = ['expert-IDLE', 'type-bug', 'invalid']
title = 'IDLE: Input method error in comment with Korean language'
updated_at = <Date 2017-07-11.23:25:15.854>
user = 'https://bugs.python.org/DelTos'
```

bugs.python.org fields:
```python
activity = <Date 2017-07-11.23:25:15.854>
actor = 'terry.reedy'
assignee = 'terry.reedy'
closed = True
closed_date = <Date 2017-07-11.23:25:15.855>
closer = 'terry.reedy'
components = ['IDLE']
creation = <Date 2017-07-10.06:33:39.665>
creator = 'DelTos'
dependencies = []
files = ['46998', '47003', '47004']
hgrepos = []
issue_num = 30890
keywords = []
message_count = 10.0
messages = ['298028', '298101', '298102', '298103', '298105', '298106', '298107', '298108', '298109', '298189']
nosy_count = 3.0
nosy_names = ['terry.reedy', 'louielu', 'DelTos']
pr_nums = []
priority = 'normal'
resolution = 'not a bug'
stage = 'resolved'
status = 'closed'
superseder = None
type = 'behavior'
url = 'https://bugs.python.org/issue30890'
versions = ['Python 3.6']
```

</p></details>

**Comment by 13961e4d-17cc-4ee4-91db-63e998679732** [comment]
Like the attached image file, there is a little error in comments.
If I write the comments with Korean, the last character is printed small and color is not applied.
It can be fixed with pressing space bar once, but I hope it is solved in the editor.

**Comment by terryjreedy** [comment]
Did you actually test on 3.7?  If you did, and got different results, then this is not an IDLE issue, as it is unchanged in 3.7.

This might be a tcl/tk issue rather than IDLE issue.

**Comment by 13961e4d-17cc-4ee4-91db-63e998679732** [comment]
It was my mistake. I tested it on IDLE 3.6.

**Comment by terryjreedy** [comment]
I can seen that the last char is not colored correctly, but not that it is small.  The white pixels seems to comprise a character that is about as big as the others.  But I don't know Hangul chars.

What version of Tk (see Help =\> About IDLE), and what OS?

Please upload your main.py so I can load it into my IDLE on Win10.

**Comment by 13961e4d-17cc-4ee4-91db-63e998679732** [comment]
I attched simple main.py file.
TK version is 8.6.6 and OS is Windows 10.

**Comment by terryjreedy** [comment]
File loads normally for me. What happens when you load it as is?

If you need to type something, can you produce effect with entering numbers with keypad, without using any special Korean entry method that I would not have?  If so, give exact directions and since we both have win10, I will try.

**Comment by 9da18a29-a07d-4828-a801-24cc911246de** [comment]
I'm using Arch Linux with HIME input method, and using 3.7a0 Python. But I can't reproduce your problem in Korean input.

Could you provide how to type in the sentence in your picture? (Roman or the stroke maybe) and your environment?

thanks.

**Comment by 13961e4d-17cc-4ee4-91db-63e998679732** [comment]
It doesn't occur when I'm typing comments with numbers or English. It only occurs in Korean comments.
Plus, I changed the font and another problem has occured. The letters are printed in vertically.(Attached file) It seems there is a problem only in the Korean language.
I gave up typing comments with Korean and changed it into English.

**Comment by 13961e4d-17cc-4ee4-91db-63e998679732** [comment]
Or it may be a problem that only happens on my PC.

**Comment by terryjreedy** [comment]
I am thinking that this is a artifact of the combination of monitor pixel and subpixel type, size, and arrangement; font type and size; particular character; and character generation software. When I load the file in IDLE, the Korean comment chars are a nice uniform red.  But the l in color is tinged green while in 'for x in range', the vertical parts of first r, i, and n are gray instead of keyword orange.  Range is spotty because builtin purple uses red and blue subpixels while green is off.

The spacebar bit through me off and still puzzles me, but overall color glitches are not limited to Korean.

Bottom line: IDLE can tell tk 'color this substring red', but has no control after that as to what happens.  So I am 99.9% sure this is not an IDLE bug.
